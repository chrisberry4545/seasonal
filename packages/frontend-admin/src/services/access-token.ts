import { getStorage } from '../consts';

export const setAccessToken = (token: string) => {
  getStorage().accessToken = token;
};

export const getAccessTokenHeaders = (): HeadersInit => {
  const token = getStorage().accessToken;
  return token
  ? {
    Authorization: `Bearer ${token}`
  }
  : {};
};
