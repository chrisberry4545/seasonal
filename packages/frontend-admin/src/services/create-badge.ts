import { IBadge } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { BADGE_URL } from '../config';

export const createBadge = async (badge: IBadge): Promise<IBadge> =>
  makeAuthorizedRequest<IBadge>(BADGE_URL, {
    body: JSON.stringify(badge),
    method: 'POST'
  });
