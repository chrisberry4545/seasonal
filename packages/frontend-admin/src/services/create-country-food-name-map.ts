import { ICountryFoodNameMap } from '@chrisb-dev/seasonal-shared';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_FOOD_NAME_MAP_URL } from '../config';

export const createCountryFoodNameMap = async (
  country: ICountryFoodNameMap
): Promise<ICountryFoodNameMap> =>
  makeAuthorizedRequest<ICountryFoodNameMap>(COUNTRY_FOOD_NAME_MAP_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
