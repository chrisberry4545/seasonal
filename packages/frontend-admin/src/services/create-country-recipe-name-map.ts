import { ICountryRecipeNameMap } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_RECIPE_NAME_MAP_URL } from '../config';

export const createCountryRecipeNameMap = async (
  country: ICountryRecipeNameMap
): Promise<ICountryRecipeNameMap> =>
  makeAuthorizedRequest<ICountryRecipeNameMap>(COUNTRY_RECIPE_NAME_MAP_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
