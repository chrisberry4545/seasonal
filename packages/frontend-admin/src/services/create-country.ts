import { ICountry } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { COUNTRY_URL } from '../config';

export const createCountry = async (country: ICountry): Promise<ICountry> =>
  makeAuthorizedRequest<ICountry>(COUNTRY_URL, {
    body: JSON.stringify(country),
    method: 'POST'
  });
