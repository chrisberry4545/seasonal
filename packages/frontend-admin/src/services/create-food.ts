import { IFood } from '@chrisb-dev/seasonal-shared';
import { makeAuthorizedRequest } from './make-authorized-request';
import { FOOD_URL } from '../config';

export const createFood = async (food: IFood): Promise<IFood> =>
  makeAuthorizedRequest<IFood>(FOOD_URL, {
    body: JSON.stringify(food),
    method: 'POST'
  });
