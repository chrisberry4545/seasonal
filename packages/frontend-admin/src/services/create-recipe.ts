import { IRecipe } from '@chrisb-dev/seasonal-shared-models';
import { makeAuthorizedRequest } from './make-authorized-request';
import { RECIPE_URL } from '../config';

export const createRecipe = async (recipe: IRecipe): Promise<IRecipe> =>
  makeAuthorizedRequest<IRecipe>(RECIPE_URL, {
    body: JSON.stringify(recipe),
    method: 'POST'
  });
