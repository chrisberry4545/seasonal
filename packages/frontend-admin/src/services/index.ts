export * from './set-access-token';

export * from './create-badge';

export * from './create-country-food-name-map';

export * from './create-country-recipe-name-map';

export * from './create-country';

export * from './create-food';

export * from './create-recipe';

export * from './create-region-food-season-map';

export * from './create-region';

export * from './create-translations-badge-name';

export * from './create-translations-country-name';

export * from './create-translations-season-name';

export * from './create-translations-region-name';

export * from './create-user';

export * from './delete-badge';

export * from './delete-country-food-name-map';

export * from './delete-country-recipe-name-map';

export * from './delete-country';

export * from './delete-food';

export * from './delete-recipe';

export * from './delete-region-food-season-map';

export * from './delete-region';

export * from './delete-translations-badge-name';

export * from './delete-translations-country-name';

export * from './delete-translations-season-name';

export * from './delete-translations-region-name';

export * from './delete-user';

export * from './get-all-badges';

export * from './get-all-countries';

export * from './get-all-country-food-name-map';

export * from './get-all-country-recipe-name-map';

export * from './get-all-food';

export * from './get-all-recipes';

export * from './get-all-region-food-season-map';

export * from './get-all-regions';

export * from './get-all-translations-badge-names';

export * from './get-all-translations-country-names';

export * from './get-all-translations-season-names';

export * from './get-all-translations-region-names';

export * from './get-all-users';

export * from './get-one-badge';

export * from './get-one-country';

export * from './get-one-country-food-name-map';

export * from './get-one-region-food-season-map';

export * from './get-one-country-recipe-name-map';

export * from './get-one-food';

export * from './get-one-recipe';

export * from './get-one-region';

export * from './get-one-translations-badge-name';

export * from './get-one-translations-country-name';

export * from './get-one-translations-season-name';

export * from './get-one-translations-region-name';

export * from './get-one-user';

export * from './login-request';

export * from './make-authorized-request';

export * from './update-badge';

export * from './update-country';

export * from './update-country-food-name-map';

export * from './update-region-food-season-map';

export * from './update-country-recipe-name-map';

export * from './update-food';

export * from './update-recipe';

export * from './update-region';

export * from './update-translations-badge-name';

export * from './update-translations-country-name';

export * from './update-translations-season-name';

export * from './update-translations-region-name';

export * from './update-user';
