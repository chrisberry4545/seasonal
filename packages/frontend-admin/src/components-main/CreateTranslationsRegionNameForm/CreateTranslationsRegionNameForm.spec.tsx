import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ITranslationsRegionName } from '@chrisb-dev/seasonal-shared-models';
import * as services from '../../services';
import {
  CreateTranslationsRegionNameForm
} from './CreateTranslationsRegionNameForm';
import {
  BaseFormTranslationsRegionName
} from '../BaseFormTranslationsRegionName/BaseFormTranslationsRegionName';

jest.mock('../../services', () => ({
  createTranslationsRegionName: jest.fn()
}));
jest.mock('../../consts', () => ({
  FORM_BUTTON_TEXT: {
    CREATE: 'create'
  }
}));
jest.mock('../../components-layouts', () => ({
  LayoutWithTitle: () => 'LayoutWithTitle'
}));
jest.mock('../BaseFormTranslationsRegionName/BaseFormTranslationsRegionName', () => ({
  BaseFormTranslationsRegionName: () => 'BaseFormTranslationsRegionName'
}));

describe('<CreateTranslationsRegionNameForm />', () => {
  let wrapper: ShallowWrapper;
  let mockCreate: jest.SpyInstance;

  beforeEach(() => {
    mockCreate = jest.spyOn(services, 'createTranslationsRegionName')
      .mockResolvedValue({} as ITranslationsRegionName);
    wrapper = shallow(
      <CreateTranslationsRegionNameForm />
    );
  });

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  test('calls create on update', () => {
    const item = {} as ITranslationsRegionName;
    const update = wrapper.find(BaseFormTranslationsRegionName).props().updateMethod;
    if (update) {
      update(item);
    }
    expect(mockCreate).toHaveBeenCalledWith(item);
  });

});
