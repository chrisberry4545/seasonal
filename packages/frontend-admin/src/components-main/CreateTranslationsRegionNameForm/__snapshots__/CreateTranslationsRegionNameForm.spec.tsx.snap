// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateTranslationsRegionNameForm /> renders correctly 1`] = `
<LayoutWithTitle
  title="Create Translations Region Name"
>
  <BaseFormTranslationsRegionName
    buttonText="create"
    items={
      Object {
        "id": "",
        "languages": Array [],
        "name": "",
        "regionId": "",
      }
    }
    updateMethod={[MockFunction]}
  />
</LayoutWithTitle>
`;
